server/appservices/service.go

```diff
@@ -33,14 +43,45 @@ type Service interface {
 	GetOAuth2User(_ *incoming.Request) ([]byte, error)
 }
 
+type Caller interface {
+	InvokeCall(*incoming.Request, apps.CallRequest) (*apps.App, apps.CallResponse)
+	NewIncomingRequest() *incoming.Request
+}
+
 type AppServices struct {
-	store *store.Service
+	store     *store.Service
+	scheduler *cluster.JobOnceScheduler
```

Comment:
Here's my biggest issue with this PR. JobOnceScheduler relies on `KVList`. plugin-apps stores all of the app keys in its keyspace (session and ouath2 tokens), so the list is going to be huge and `KVList` slow.

2/5 consider https://github.com/mattermost/mattermost-server/pull/21653

Reply:
For reference: `cluster.JobOnceScheduler` calls `KVList` on `Start()` and every 5 minutes.

Reply:
Could you please elaborate on how a solution that uses https://github.com/mattermost/mattermost-server/pull/21653 would work?

I would assume only the cluster leader executes timers, correct? Once a plugin instance received the request to create a new timer, it needs to communicate that information to the leader. How would that work?

Reply:
I've talked with @levb offline. The solutions in https://github.com/mattermost/mattermost-plugin-apps/pull/375#issuecomment-1410445150 are the path forward, but require the cache store from https://github.com/mattermost/mattermost-plugin-apps/pull/458.